### ISSUE TITLE
cfg_attr condition is ignored

### ISSUE THREAD

**Issue by RReverser** [rust-analyzer/rust-analyzer#4707]
Take a look at this simplest `lib.rs`:

```rust
#![cfg_attr(feature = "nightly", feature(never_type))]

pub fn foo() {}
```

Cargo.toml:

```toml
[package]
name = "ra-test"
version = "0.1.0"
authors = ["Ingvar Stepanyan <me@rreverser.com>"]
edition = "2018"

[features]
default = []
nightly = []
```

Workspace settings:

```json
{
	"rust-analyzer.cargo.allFeatures": false
}
```

I'm getting an error on the first line of `lib.rs` saying

> `#![feature]` may not be used on the stable release channel rustc(E0554)

This `feature` should never be activated, because it's not explicitly set and `allFeatures` is also `false`. I also think that I didn't see this error on the same condition in the past (around ~2 months ago), but hard to tell for sure.

**Comment by RReverser** [comment]
Ah, right after submitting I finally realised what's happening - that error is not from LSP but from the `cargo check` that Rust Analyzer invokes and I need to disable `"rust-analyzer.checkOnSave.allFeatures": false` separately as well.

To be honest, to me as a user this seems fairly confusing and should be an implementation detail IMO - that is, one setting should control Cargo features for LSP, `cargo check` and any other invocation.

**Comment by bjorn3** [comment]
You may want rust-analyzer to analyze and show all types for code behind feature flags without compiling it with those feature flags enabled.

**Comment by RReverser** [comment]
Idk, it seems like a footgun, especially given that both errors look exactly the same when you're used to saving after every small change. Most of the time, as a user, I'd want to have confidence that if something works in my editor, it will work when compiled, too, and vice versa.

Maybe there's a use-case for opt-outs to achieve behaviour like you're describing, but I'd suggest it should be at least a very explicit opt-out in the config.

**Comment by bjorn3** [comment]
Maybe the `cargo` part of `rust-analyzer.cargo.allFeatures` should be renamed to something else?

**Comment by lnicola** [comment]
See also #4704.

**Comment by lnicola** [comment]
Closed by #4711
